todo_app/auth/decorators.py

```diff
@@ -0,0 +1,14 @@
+from functools import wraps
+from flask_login import current_user
+
+from todo_app.auth.user_roles import UserRoles
+
+
+def requireWriter(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        if current_user.is_anonymous or current_user.role == UserRoles.WRITER:
+            return func(*args, **kwargs)
+        else:
+            return {"message": "Unauthorized"}, 401
```

Comment:
Just saw another learner choosing to make these all 404 results rather than 401 and I thought that was interesting - [this article gives a decent run down for when you might want to make that call](https://www.bennadel.com/blog/2400-handling-forbidden-restful-requests-401-vs-403-vs-404.htm#:~:text=The%20three%20status%20codes%20that,404%20%2D%20Not%20Found)